dashboard/app.py

```diff
@@ -22,3 +22,26 @@
 
 if selected_event_type == EventTypes.INPUT_EVENTS.value:
     pass
+
+# Aggregate the visit data
+aggregated_data = get_aggregated_visit_data(visit_events)
+
+# Create a dropdown for page selection
+pages = aggregated_data['pageUrl'].unique()
+selected_page = st.selectbox('Select a page to analyze:', ['All'] + list(pages))
+
+# Filter data based on the selected page
+if selected_page != 'All':
+    filtered_data = aggregated_data[aggregated_data['pageUrl'] == selected_page]
+else:
+    filtered_data = aggregated_data
+
+# Ensure the data is sorted by date for consistent plotting
+filtered_data = filtered_data.sort_values('date')    
+
+# Plotting
+st.line_chart(data=filtered_data,
+              x='date',
+              y='number of visitors',
+              color='pageUrl' if selected_page == 'All' else None,
+              use_container_width=True)
```

Comment:
This looks good - but could you put this in a method in `dashboard/widgets/visit_event_widgets.py` and import that method into this file